### PR TITLE
Add cleanup for namespaces on local cluster delete

### PR DESCRIPTION
**Pull request by dramich**
Issue: https://github.com/rancher/rancher/issues/16602

### CODE REVIEW

File: pkg/controllers/management/usercontrollers/controller.go [hunk 1]
```diff
@@ -317,6 +347,75 @@ func (c *ClusterLifecycleCleanup) updateClusterRoleBindingOwner(
...
+func cleanupNamespaces(client kubernetes.Interface) error {
+	logrus.Debug("Starting cleanup of local cluster namespaces")
+	namespaces, err := client.CoreV1().Namespaces().List(metav1.ListOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, ns := range namespaces.Items {
+		err = tryUpdate(func() error {
+			nameSpace, err := client.CoreV1().Namespaces().Get(ns.Name, metav1.GetOptions{})
```
Comment (StrongMonkey) [inline pkg/controllers/management/usercontrollers/controller.go]: can you use ns without calling `get` again?

File: pkg/controllers/management/usercontrollers/controller.go [hunk 2]
```diff
@@ -317,6 +347,75 @@ func (c *ClusterLifecycleCleanup) updateClusterRoleBindingOwner(
...
+				}
+				return err
+			}
+
+			var updated bool
+
+			// Cleanup finalizers
+			if len(nameSpace.Finalizers) > 0 {
+				finalizers := []string{}
+				for _, finalizer := range nameSpace.Finalizers {
```
Comment (StrongMonkey) [inline pkg/controllers/management/usercontrollers/controller.go]: i think `range nameSpace.Finalizers` should take care of the zero length case.

Comment (dramich) [reply pkg/controllers/management/usercontrollers/controller.go]: I added the check to stop the allocation and the check on 377 if there aren't any

**Comment by StrongMonkey** [comment]
LGTM

**Pull request merged**